### DISCUSSION TITLE
Does the search function return a reference to db's record in-memory?

### DISCUSSION THREAD

**Discussion by jingjiezhu** [msiemens/tinydb#515] [Q&A]
A snippt codes
```
from tinydb import TinyDB, Query
db =  TinyDB("test.json")
db.insert({"id": 1})

q = Query()
records = db.search(q.id == 1)
records[0]["name"] = "N1"

db.search(q.id==1)
>>[{'id': 1, 'name': 'N1'}]
db.all()
>>[{'id': 1}]
```

You can see that modifications to the search function result affect the later search results.  But it doesn't change the records on the disk. Is it expected or not?

**Comment by msiemens** [comment]
The behavior you get seems to relate to the query cache feature built into TinyDB. It's definetly unexepcted, I'll turn this into an issue in TinyDB altough I'm not sure how to best deal with this.
